File: Algorithms/p007_Reverse_Integer/p007_Reverse_Integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long res = 0;
        while(x){
            res = res*10+x%10;
            x /= 10;
        }
        return (res>=numeric_limits<int>::min()&&res<=numeric_limits<int>::max()) ? res : 0;
    }
};

File: Algorithms/p007_Reverse_Integer/p007_Reverse_Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        res = 0
        x1 = abs(x)
        while x1:
            res = res*10+x1%10
            x1 //= 10
        return (res if x>=0 else -res) if res>=-2147483648 and res<=2147483647 else 0
